### PR TITLE
Rails 7 support

### PR DESCRIPTION
**Pull request by DmitryTsepelev**

### CODE REVIEW

**Comment by DmitryTsepelev** [comment]
GitHub Actions are flaky today, so everything will be eventually green.

@djezzzl could you please try it out as well?

File: lib/ar_lazy_preload/contexts/base_context.rb [hunk 1]
```diff
@@ -75,9 +81,9 @@ def loaded_association_names
         @loaded_association_names ||= Set.new
       end
 
-      def preloader
-        @preloader ||= ActiveRecord::Associations::Preloader.new
-      end
+      # def preloader
+      #   @preloader ||= ActiveRecord::Associations::Preloader.new
+      # end
```
Comment (djezzzl) [inline lib/ar_lazy_preload/contexts/base_context.rb:86]: **Note:** Let's remove this?

File: lib/ar_lazy_preload/active_record/relation.rb [hunk 2]
```diff
@@ -82,6 +80,16 @@ def lazy_preload_values
...
     private
 
+    def perform_preloading(records, associations)
+      if ::ActiveRecord::VERSION::MAJOR >= 7
+        ActiveRecord::Associations::Preloader.new(
+          records: records, associations: associations
+        ).call
+      else
+        ActiveRecord::Associations::Preloader.new.preload(records, associations)
+      end
```
Comment (djezzzl) [inline lib/ar_lazy_preload/active_record/relation.rb:90]: **Question:** Maybe it's easier/better to have 2 different classes for version 7 and below? So we can include in appropriate version only what we need.

File: lib/ar_lazy_preload/contexts/base_context.rb [hunk 3]
```diff
@@ -63,7 +63,13 @@ def perform_preloading(association_name)
...
       def preload_records(association_name, records)
         TemporaryPreloadConfig.within_context do
-          preloader.preload(records, association_name)
+          if ::ActiveRecord::VERSION::MAJOR >= 7
+            ActiveRecord::Associations::Preloader.new(
+              records: records, associations: [association_name]
+            ).call
+          else
+            ActiveRecord::Associations::Preloader.new.preload(records, association_name)
+          end
```
Comment (djezzzl) [inline lib/ar_lazy_preload/contexts/base_context.rb:72]: **Note:** Same here as above.

**Review by djezzzl — APPROVED** [review]

**Comment by djezzzl** [comment]
Couldn't test it within https://github.com/djezzzl/n1_loader as it requires more things to be done before but if specs are green, then I'd love to see this as released 👍

**Pull request merged**